src/services/dashboards/dashboards.svc.coffee

```diff
@@ -110,26 +110,47 @@ angular
             (success)->
               orgId = success.currentOrganization.id
 
+              # Retrieve dashboards
               $http.get(ImpacRoutes.dashboards.index(orgId)).then(
                 (dashboards)->
                   _self.setDashboards(dashboards.data).then(->
                     _self.setCurrentDashboard()
                     deferred.resolve(_self.config)
                     $log.info("Impac! - DashboardsSvc: loaded (force=#{force})")
 
-                  ).finally( -> _self.loadLocked=false )
+                  ).finally( -> _self.loadLocked = false )
                 (error)->
                   $log.error("Impac! - DashboardsSvc: cannot retrieve dashboards list for org: #{orgId}")
                   _self.loadLocked=false
                   deferred.reject(error)
               )
+
+              # Retrieve widgets templates
+              url = 
+                if ImpacTheming.get().dhbWidgetsConfig.templates.defaultToFinancialYear
+                  fy_end_month = ImpacMainSvc.getFinancialYearEndMonth()
+                  "#{ImpacRoutes.widgets.templates()}?financial_year_end_month=#{fy_end_month}"
+                else
+                  ImpacRoutes.widgets.templates()
+
+              $http.get(url).then(
+                (response) ->
+                  _self.setWidgetsTemplates(response.data.widgets)
+                  _self.loadLocked = false
+
```

Comment:
Hmm I'm really not sure if this method will run consistently. `@load` returns a deferred, so one would expect this success handler here to `deferred.resolve()` something... But the get dashboards call above success handler has a deferred resolve. Not really a good idea to have two success handlers like this, the code invoking this function will get some pretty weird responses should it define success / error handlers. 

Maybe you should consider using $q.all ? Or moving this into another method that gets called on dashboards load success?